tests/unit/test_charm.py

```diff
@@ -95,50 +97,67 @@ def test_replica_consensus_stable_after_leader_reelection(
 
 
 @pytest.mark.usefixtures("attach_storage")
-def test_mysql_relation(
-    harness: ops.testing.Harness, setup_db_relation: typing.Callable[[], typing.Tuple[int, dict]]
+def test_db_relation(
+    harness: ops.testing.Harness,
+    setup_db_relation: typing.Callable[[], typing.Tuple[int, dict]],
 ):
     """
     arrange: no pre-condition.
-    act: add and remove the database relation between WordPress application and mysql.
+    act: add and remove the db relation between WordPress application and mysql.
     assert: database info in charm state should change accordingly.
     """
+    harness.begin_with_initial_hooks()
+    charm: WordpressCharm = typing.cast(WordpressCharm, harness.charm)
 
-    def get_db_info_from_state():
-        """Wrapper for getting database relation state information.
+    assert (
+        charm._current_effective_db_info is None
+    ), "database info in charm state should not exist before database relation created"
 
-        Returns:
-            Wrapped dictionary of database relation information.
-        """
-        return {
-            "host": charm.state.relation_db_host,
-            "database": charm.state.relation_db_name,
-            "user": charm.state.relation_db_user,
-            "password": charm.state.relation_db_password,
-        }
+    db_relation_id, db_info = setup_db_relation()
+
+    effective_db_info = charm._current_effective_db_info
+
+    assert effective_db_info.hostname == db_info["host"]
+    assert effective_db_info.database == db_info["database"]
+    assert effective_db_info.username == db_info["user"]
+    assert effective_db_info.password == db_info["password"]
+
+    harness.remove_relation(db_relation_id)
```

Comment:
What is the reason for needing to do this please? It looks a bit like cleanup which would be more appropriate in a fixture?

Reply:
This is needed to test the behavior after removing the relation, to verify that the charm goes back into the BlockedState, rather than to clean up :)